src/debounce.jl

```diff
@@ -0,0 +1,224 @@
+### A Pluto.jl notebook ###
+# v0.18.0
+
+using Markdown
+using InteractiveUtils
+
+# ╔═╡ d0379e1e-8a6b-11ec-0ae6-3506c99b8004
+import PlutoLinks, PlutoHooks
+
+# ╔═╡ 33e53084-ddca-4147-ae09-fd9c0deec242
+"""
+Run a code block using debounce, i.e. only run it when variable list has stabilized through `timeout`.
+
+```julia
+@use_debounce([var], 1) do
+	sleep(2.)
+	@info "stable variable \$var"
+end
+```
+"""
+macro use_debounce(f, deps, timeout)
+	quote
+		let
+			value, set_value = @PlutoLinks.use_state(nothing)
+
+			count_ref = @PlutoHooks.use_ref(0)
+			count_ref[] += 1
+
+			@PlutoLinks.use_task($(esc(deps))) do
+				sleep($timeout)
```

Comment:
I think it should also be escaped if we want to have timeout specified by a variable `@use_debounce(f, [x,], y)`
```suggestion
				sleep($(esc(timeout)))
```